pandas/core/arrays/datetimes.py

```diff
@@ -735,14 +737,35 @@ def astype(self, dtype, copy: bool = True):
     def _format_native_types(
         self, *, na_rep: str | float = "NaT", date_format=None, **kwargs
     ) -> npt.NDArray[np.object_]:
-        from pandas.io.formats.format import get_format_datetime64_from_values
-
-        fmt = get_format_datetime64_from_values(self, date_format)
+        if date_format is None:
+            ido = self._is_dates_only
+            if ido:
+                # Only dates and no timezone: provide a default format
+                date_format = "%Y-%m-%d"
```

Comment:
```suggestion
        if date_format is None and self._is_dates_only:
            # Only dates and no timezone: provide a default format
            date_format = "%Y-%m-%d"
```

Nit

Reply:
good idea, will update

Reply:
#55405 implements this same method in the mixin class TimelikeOps.  That is the better long-term location.  Will update whichever PR goes in second.